### ISSUE TITLE
Something is wrong with historical timezone changes.

### ISSUE THREAD

**Issue by Reggino** [moment/moment-timezone#345]
Thanks for this, awesome work!

I did however run into an issue: Considers the following code:

```
var moment = require('moment-timezone');

var date1 = moment('1940-05-16T02:00:00+02:00');
console.log(date1.format());
console.log(date1.startOf('day').format());

var date2 = moment('1940-05-16T00:00:00.000Z');
date2.tz('Europe/Amsterdam');
console.log(date2.format());
console.log(date2.startOf('day').format());

```

Expected output:

```
1940-05-16T02:00:00+02:00
1940-05-16T00:00:00+02:00
1940-05-16T02:00:00+02:00
1940-05-16T00:00:00+02:00

```

Actual output (in latest version):

```
1940-05-16T02:00:00+02:00
1940-05-16T00:00:00+02:00
1940-05-16T02:00:00+02:00
1940-05-15T23:20:00+00:20
```

1940-05-16 is a special date: this day the timezone changed from +00:20 to +02:00, See [Wikipeda](https://en.wikipedia.org/wiki/UTC%2B00:20). It looks like `startOf` doesn't honor the timezone change correctly after being adjusted to the Europe/Amsterdam timezone?

**Comment by Reggino** [comment]
To make things more complicated, see 
[this page](http://www.timeanddate.com/time/change/netherlands/amsterdam?year=1940) with more details on the change. DST started on that exact date. So on __00:00:00__ the clock turned forward 1:40 hour .

**Comment by Reggino** [comment]
The same issue applies to other history related timezone changes. On 1937-07-01 the timezone changed from +01:19 to +01:20 in Europe/Amsterdam.

```
var moment = require('moment-timezone');

var date2 = moment('1937-07-01T00:00:00.000Z');
date2.tz('Europe/Amsterdam');
console.log(date2.format());
date2.startOf('day');
console.log(date2.format());
```

has output

```
1937-07-01T01:20:00+01:20
1937-06-30T23:59:00+01:19
```
Where i expected
```
1937-07-01T01:20:00+01:20
1937-06-01T00:00:00+01:20
```

**Comment by mj1856** [comment]
Yes, it appears there is something going on here.

```javascript
moment.tz('1940-05-15', 'Europe/Amsterdam').format()
// "1940-05-15T00:00:00+00:20"  (correct)
moment.tz('1940-05-15', 'Europe/Amsterdam').startOf('day').format()
// "1940-05-15T00:00:00+00:20"  (correct)
moment('1940-05-15T00:00:00Z').tz('Europe/Amsterdam').format()
// "1940-05-15T00:20:00+00:20"  (correct)
moment('1940-05-15T00:00:00Z').tz('Europe/Amsterdam').startOf('day').format()
// "1940-05-15T00:00:00+00:20"  (correct)

moment.tz('1940-05-16', 'Europe/Amsterdam').format()
// "1940-05-16T01:40:00+02:00"  (correct)
moment.tz('1940-05-16', 'Europe/Amsterdam').startOf('day').format()
// "1940-05-16T02:00:00+02:00"  (incorrect - should be "1940-05-16T01:40:00+02:00")
moment('1940-05-16T00:00:00Z').tz('Europe/Amsterdam').format()
// "1940-05-16T02:00:00+02:00"  (correct)
moment('1940-05-16T00:00:00Z').tz('Europe/Amsterdam').startOf('day').format()
// "1940-05-15T23:20:00+00:20"  (incorrect - should be "1940-05-16T01:40:00+02:00")
```

**Comment by mj1856** [comment]
Related issues:  #327, moment/moment#3132, moment/moment#2749

**Comment by gilmoreorless** [comment]
This was fixed in `moment v2.24.0`, as long as you're using a version of `moment-timezone` that includes the full 1940 dataset.
